src/main/java/Storage.java

```diff
@@ -0,0 +1,84 @@
+import task.type.Deadline;
+import task.type.Event;
+import task.type.Task;
+import task.type.Todo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Storage {
+    private static ArrayList<Task> taskList = new ArrayList<>();
+    private static final String PATH = "data/savedData.txt";
+    private static final String DIRECTORY = "data";
+
+    public static void saveData(ArrayList<Task> tasks) throws IOException{
+        StringBuilder line = new StringBuilder();
+        for(Task task: tasks){
+            line.append(serializeTask(task));
+        }
+
+        FileWriter fw = new FileWriter(PATH);
+        fw.write(line.toString());
+        fw.close();
+    }
+
+    public static ArrayList<Task> loadData() throws FileNotFoundException{
+        File directory = new File(DIRECTORY);
+        if(!directory.exists()){
+            if(directory.mkdir()){
+                System.out.println("Successfully created a new directory for save file");
+            }
+        } else {
```

Comment:
You can make the [happy path](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#make-the-happy-path-prominent) more prominent here. After checking if that the directory doesn't exist and making the new directory, you can immediately `return`, then the rest of the logic does not need to be nested inside the `else`